### PR TITLE
Testsets for sparse vector tests

### PR DESCRIPTION
**Pull request by kshyatt**
@Sacha0 please verify that I have retained the letter and spirit of your comprehensive and lovely tests :)

### CODE REVIEW

File: test/sparse/sparsevector.jl [hunk 1]
```diff
@@ -24,24 +23,20 @@ let x = spv_x1
...
 end
 
-@test count(SparseVector(8, [2, 5, 6], [true,false,true])) == 2
-
-# full
-
-for (x, xf) in [(spv_x1, x1_full)]
-    @test isa(Array(x), Vector{Float64})
-    @test Array(x) == xf
+@testset "full/Array" begin
```
Comment (fredrikekre) [inline test/sparse/sparsevector.jl]: Rename to "Array constructor" perhaps?

File: test/sparse/sparsevector.jl [hunk 2]
```diff
@@ -24,24 +23,20 @@ let x = spv_x1
...
+    for (x, xf) in [(spv_x1, x1_full)]
```
Comment (fredrikekre) [inline test/sparse/sparsevector.jl]: What a useless for-loop 😄

File: test/sparse/sparsevector.jl [hunk 3]
```diff
@@ -24,24 +23,20 @@ let x = spv_x1
...
+        @test isa(Array(x), Vector{Float64})
+        @test Array(x) == xf
```
Comment (fredrikekre) [inline test/sparse/sparsevector.jl]: Perhaps also test `Vector(x)`

**Comment by Sacha0** [comment]
> please verify that I have retained the letter and spirit of your comprehensive and lovely tests :)

Flattered as I am, such kind words should be directed to those who wrote the vast majority of these tests / the associated functionality (e.g. `lindahua`, `mbauman`, `viralbshah`, `tanmayk`, `andreasnoack`, and others) rather than me :).

File: test/sparse/sparsevector.jl [hunk 4]
```diff
@@ -255,326 +249,326 @@ let x = SparseVector(10, [2, 7, 9], [2.0, 7.0, 9.0])
...
+        @test x.nzind !== xc.nzval
+        @test x.nzval !== xc.nzval
+        @test exact_equal(x, xc)
+    end
 
-    xc = convert(SparseVector{Float32,Int}, x)
-    xf32 = SparseVector(8, [2, 5, 6], [1.25f0, -0.75f0, 3.5f0])
-    @test isa(xc, SparseVector{Float32,Int})
-    @test exact_equal(xc, xf32)
+    let a = SparseVector(8, [2, 5, 6], Int32[12, 35, 72])
```
Comment (Sacha0) [inline test/sparse/sparsevector.jl]: This block of tests does not appear associated with `copy[!]`?

**Comment by kshyatt** [comment]
What's up with CI? Are we happy with this otherwise?

File: test/sparse/sparsevector.jl (hunk 2 above)
Comment (fredrikekre) [reply test/sparse/sparsevector.jl]: Remove this loop? :)

**Comment by kshyatt** [comment]
OK CI has passed - is this ok to merge or no?

**Pull request merged**